### PR TITLE
Bug fixes and enchancements in integration

### PR DESCRIPTION
**Pull request by cyble-dev**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by RosenbergYehuda — COMMENTED** [review]
Hello @cyble-dev,
I would like to start reviewing your contribution, but I noticed that you haven't filled out the Contribution Form. Please complete the form so that I can proceed with the review. You can locate the form in previous commits.
Thank you!

File: Packs/CybleEventsV2/ReleaseNotes/1_0_2.md
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/ReleaseNotes/1_0_2.md:1]: I'm not sure why you have 2 release notes.
So please remove the extra one, and fill out the release notes with what you added to this version.

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda,
I have filled the form, you may proceed.
Thanks.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md:17]: I think it's better not to use personal pronouns ("you", "check the box").
can you think of something else?

**Review by RosenbergYehuda — COMMENTED** [review]
Hello @cyble-dev, Great work!
I wanted to provide you with an update as we enter the holiday season here in Israel. Please expect some potential delays from our end during this time.

I've begun my code review, and I'll finish it at the beginning of next week. In the meantime, I'll send you some minor comments I've noted. If you require further clarification or anything else from me, please don't hesitate to reach out. 

Have a wonderful weekend!

**Comment by melamedbn** [comment]
Hi @cyble-dev,

I've taken a moment to review your content, and here are a few suggestions to enhance its alignment with our standards:

## **Playbook: Cyble Vision Alert V2**:

1. **Single Task Playbook**: The playbook currently has just one task. Our general approach leans towards playbooks encompassing multiple actions. Could you shed some light on the intent behind this structure?
2. **Hardcoded Values**: Kindly adjust the playbook inputs by removing the hardcoded values. Illustrating with examples in the input descriptions might be more apt.
3. **'Done' Header**: Please introduce a 'Done' section header in line with our guidelines.
4. **Playbook Image Naming**: There's a minor tweak needed in the `doc_files` folder. Could we rename the playbook image by eliminating the date?

## **Classifier**:

1. **Incident Type Mapping**: The current mapping directs to the 'CybleEventsv2' incident type, which seems absent. Could we review this?
2. **Context Key Selection**: A suggestion for better efficacy - consider employing the complex context method when picking context keys.

## **Layout**:

1. **Cyble Event v2 Specifics**: For clarity, could you detail any distinct aspect of the Cyble Event v2 layout? If there are specialized Cyble fields or sections, understanding their rationale would be beneficial.

Your commitment to quality is commendable. These suggestions are aimed to further polish the content, ensuring it's in perfect sync with our best practices. I'm here to assist, so don't hesitate to reach out for any clarifications or further discussions.

Warm regards,

Ben

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml:189]: isn't this only for mapping in? you added only outgoing mapper. is that correct?

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:328]: in the .yml file the default is 1. why is it 0 here?

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:322]: in the yml file the default value is 'desc', it should be the same.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:326]: Please remove.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:423]: Please add a doc string to the new functions you added, explaining what the functions do, what does it expect as input and output.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:484]: I think the address should be as a constant at the top of the code, that way if it's changed it will be easy to find.

File: Packs/CybleEventsV2/pack_metadata.json
Comment (RosenbergYehuda) [inline Packs/CybleEventsV2/pack_metadata.json:5]: this should be 1.0.2

**Review by RosenbergYehuda — CHANGES_REQUESTED** [review]
Hello @cyble-dev,

Apologies for the delayed response. I have completed my code review and added some comments - please take a look and let me know if you have any questions or disagree with anything.

I noticed you added new mirroring functionality, however there are no tests covering this new code. Could you please add unit tests for the new functions you introduced? Testing is an important part of our process to ensure quality and maintainability.

Please let me know if you need any clarification or have trouble updating the tests. I'm happy to help explain anything that is unclear. Thank you again for your contribution!

Regards,Yehuda

**Comment by RosenbergYehuda** [comment]
Hello @cyble-dev,
I want to mention that due to the holiday season in Israel, our team will be out of the office until next Tuesday the 26th. Please feel free to address the comments at your convenience, and I will review again when I'm back.

Thank you again for your contribution, I look forward to getting it merged once I return from the holiday break. Enjoy your weekend!

File: Packs/CybleEventsV2/ReleaseNotes/1_0_2.md
Comment (cyble-dev) [reply Packs/CybleEventsV2/ReleaseNotes/1_0_2.md:1]: Done

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/README.md:17]: Done

**Comment by cyble-dev** [comment]
Hello @melamedbn,
I went through your suggestions, following is our response to them.

## Playbook: Cyble Vision Alert V2:
1. **Single Task Playbook**: This is because initially we had several more functionality for the playbook, but we are working on adding a few enhancements on them. We will be adding more functionalities very soon.
2. **Harcoded Values**: Sure, I will remove the hardcoded values.
3. **'Done' Header**: Can you please share more information on where and how to do this?
4. **Playbook Image Naming**: Done

## Classifier:
1. **Incident Type Mapping**: Can you please pin point where is the current mapping directing to 'CybleEventsv2'?
2. **Context Key Selection**: Sure, will look into this.

## Layout:
1. **Cyble Event v2 Specifics**: As mentioned above, we are adding more functionalities, so will be needing a layout. As of now, it is required for the mappers.
2. Sure, will move it accordingly.

Please let me know your views on this, looking forward to your reply.
Thanks Ben.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:81]: I am raising any exception because there isn't any need to catch it. If any error occurs, I just want the code to try again, that's why I have simply caught the error and allowed the code to pass.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.yml:189]: Yes, currently only the outgoing mapping is functional. Though I have added `isremotesyncin: true` too, soon we will be adding bi-directional mapping as well.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:328]: Fixed it.

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:322]: Fixed it

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:326]: removed the commented code

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:423]: Done

File: Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py
Comment (cyble-dev) [reply Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py:484]: Yes, fixed it.

File: Packs/CybleEventsV2/pack_metadata.json
Comment (cyble-dev) [reply Packs/CybleEventsV2/pack_metadata.json:5]: Fixed

**Comment by cyble-dev** [comment]
Hello @RosenbergYehuda,
Thank you for your detailed review suggestions. I have implemented all the changes you and @melamedbn have suggested and posted a response to each of your comments.

Thanks for notifying me regarding the holiday season. We are hoping to get the PR merged ASAP so let us know once you are back and if any other changes are required.

Happy Holidays ✌️

**Comment by cyble-dev** [comment]
Hi @melamedbn,
I have one query that I couldn't find any solution for.

We need to set the value of a field from a dropdown, i.e., in the incident, to set the value of a field, there will be a dropdown. The user will select any one value from the dropdown and the field's value will be set. The values of the dropdown can't be hardcoded as they are coming from the remote server.
We want to do this as much as possible from the code, without having to configure anything manually from XSOAR UI.

Please guide me how can I do this.

Thanks.

File: Packs/CybleEventsV2/ReleaseNotes/1_0_2.md
Comment (cyble-dev) [reply Packs/CybleEventsV2/ReleaseNotes/1_0_2.md:1]: Done

**Comment by melamedbn** [comment]
> Hi @melamedbn, I have one query that I couldn't find any solution for.
> 
> We need to set the value of a field from a dropdown, i.e., in the incident, to set the value of a field, there will be a dropdown. The user will select any one value from the dropdown and the field's value will be set. The values of the dropdown can't be hardcoded as they are coming from the remote server. We want to do this as much as possible from the code, without having to configure anything manually from XSOAR UI.
> 
> Please guide me how can I do this.
> 
> Thanks.

Hi @cyble-dev,

1. As for the incident type, please remove the '-IncidentType' from both the name and id of the incident type, don't forget to adjust the classifier accordingly.
image
2. Done header: Add a task and check the 'section header' on the top right.
image
3. Dropdown issue: let's schedule a quick demo so I'll be able to provide the best solution.

Best regards,
Ben

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda,
Hope you had a wonderful holiday.
I have made all the changes as suggested by you, including the Release notes.
Please check so we can move forward.

Thanks.

**Review by RosenbergYehuda — COMMENTED** [review]
Hello @cyble-dev,

I noticed you added content to the release notes that does not follow our standard format. Please refer to our [official documentation](https://xsoar.pan.dev/docs/documentation/release-notes) on how to write release notes, or review other release notes in the repository for examples.

Specifically, the following lines can be removed as they do not impact the client:
- Added test for mirroring functionality
- Updated Readme for new features
- Created dummy_data json for update-incident functionality
- Updated release notes and version

Please let me know if you have any other questions! I'm happy to help ensure the release notes follow our standards before merging.

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda,
I have made the changes suggested by you. Please check so we can proceed.
Thanks

**Review by RosenbergYehuda — APPROVED** [review]
Great Job @cyble-dev !

I will approve this PR.

Now we will wait for @melamedbn to approve.

**Comment by cyble-dev** [comment]
Hello @melamedbn,
1. I have removed the incidentType prefix as suggested by you.
2. Pardon me, but I still don't understand where should I introduce 'Done' header. Is this in my XSOAR integration? If yes, then how and where? Or in this repo. Please suggest.
3. Sure, I'd love to. Let me know where shall I send an invite to you?

Please let me know if anything else needed to be done, so we can go ahead and merge the PR.

Thanks.

**Comment by RosenbergYehuda** [comment]
Hello @cyble-dev,

I wanted to let you know that I will be on vacation next week due to the holidays here in Israel. I will return to work on January 8th.

Please accept my apologies for any delays this may cause. 
Looking forward to catching up when I'm back in the office. 

Regards,Yehuda

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda,
We will be merging the PR before going on vacation, right? Because keeping PR on hold till January will become very problematic for us.

**Comment by RosenbergYehuda** [comment]
My bad, I'm on vacation until the 8th of October.

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda,
Oh, okay. But still, 8th October will still be very late as I am committed to a deadline. This particular enhancement has already taken longer than it should have. Please let me know how can we merge this PR within2 days.

**Comment by RosenbergYehuda** [comment]
I understand this is important, so I will make an exception and merge this while on vacation. As soon as @melamedbn approves this, please ping me and I will take care of merging it. No worries!

**Comment by RosenbergYehuda** [comment]
To get this merged as soon as possible, please ensure you address all of @ben's comments to get his approval.

Also, since I will be on vacation we won't be able to schedule a live demo. Please record a video demo of the new features you added, explaining the purpose and value of each and how it will look from the end user's perspective.

You can attach the video here as a comment, or share it via Google Drive with me at yrosenberg@paloaltonetworks.com.

Providing the demo video will help expedite the process while I'm away. Let me know if you have any other questions!

**Comment by cyble-dev** [comment]
Thank You so much for your consideration @RosenbergYehuda .
Yes, I have been mentioning @melamedbn in the comments for his approval.

The live demo he and I have mentioned is regarding that I need more clarity on adding dropdown functionality. It's not implemented yet and it's something I can't seem to figure out how to implement.
We can discuss about it after this PR is merged.

Surely I'll ping you once @melamedbn has approved the changes.

Thank You once again.

**Comment by cyble-dev** [comment]
> Hello @melamedbn,
> 
> 1. I have removed the incidentType prefix as suggested by you.
> 2. Pardon me, but I still don't understand where should I introduce 'Done' header. Is this in my XSOAR integration? If yes, then how and where? Or in this PR? Please suggest.
> 3. Sure, I'd love to. Let me know where shall I send an invite to you?
> 
> Please let me know if anything else needed to be done, so we can go ahead and merge the PR.
> 
> Thanks.

Hi @melamedbn,
Gentle reminder

**Comment by edik24** [comment]
Hi @cyble-dev can we schedule a demo for tomorrow? can you share your availability (times and timezone) and your email?

**Comment by cyble-dev** [comment]
Hi @edik24,
Can we schedule the demo at: 3rd October, 09:00am UTC?
Let me know your availability.

**Comment by edik24** [comment]
> Hi @edik24, Can we schedule the demo at: 3rd October, 09:00am UTC? Let me know your availability.

Yes it can work.
Can you share your email?

**Comment by cyble-dev** [comment]
Hi @edik24,
It's developers@cyble.com

**Comment by edik24** [comment]
Hi @cyble-dev,
9 UTC is not going to work tomorrow.

Can you suggest other times, tomorrow or day after tomorrow?

**Comment by cyble-dev** [comment]
Hi @edik24,
Can we schedule the demo at: 4th October, 09:00am UTC?
My email id: priyanshu.shrivastava@cyble.com

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda, @edik24
I have been trying to reach @melamedbn for his inputs and approval but he has not responded yet. Is there a way you can ping him to respond, as I need to merge this PR asap.
Thanks.

**Comment by melamedbn** [comment]
> Hi @RosenbergYehuda, @edik24 I have been trying to reach @melamedbn for his inputs and approval but he has not responded yet. Is there a way you can ping him to respond, as I need to merge this PR asap. Thanks.

Hi @cyble-dev,
As we discussed, I'm waiting for the demo to understand better the issues you've encountered, On that meeting I'll explain the 'Done' header.

Best regards,
Ben

**Comment by edik24** [comment]
@cyble-dev I tried also to send email to the mail you provided.
Can you please suggest some times for today or tomorrow so we can schedule a demo?

**Comment by cyble-dev** [comment]
Hi @edik24,
As mentioned above, can we schedule the demo at: 4th October, 09:00 am UTC?
You can share the invite at: priyanshu.shrivastava@cyble.com
Let me know if you face any issue in sharing the invite.

Sure @melamedbn, we will discuss at the demo call. Sounds good.

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda, @melamedbn
Thank You for merging the PR.
I have one request, I need to add a few small changes for layout modification in this PR. Will it be possible for us to add these layout changes to this PR itself? It will be a huge help and will save us a lot of time.

**Comment by cyble-dev** [comment]
> Hi @RosenbergYehuda, @melamedbn Thank You for merging the PR. I have one request, I need to add a few small changes for layout modification in this PR. Will it be possible for us to add these layout changes to this PR itself? It will be a huge help and will save us a lot of time.

Hi @RosenbergYehuda, @melamedbn
Let me know if we can do that. I will commit the updates.

**Comment by RosenbergYehuda** [comment]
Hello @cyble-dev 

Your PR has already been merged into our repo, and the original external PR is now closed, so you cannot add any additional changes to it.

I hope to finish working on it today. After that, you can open a new contribution with your proposed updates.

Are these additional changes critical? Do you want me to wait until you contribute the changes in a new PR, and then go through the review and demo process again before merging both PRs together?

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda,
No, The additional changes are not critical.
You can go ahead and close the current PR.

I will create a new PR for these additional changes later.
Thanks

**Comment by cyble-dev** [comment]
Hi @RosenbergYehuda, can you please tell me when can I expect the updated app to be live on marketplace?

**Comment by RosenbergYehuda** [comment]
Unfortunately I did not realize there were multiple unit test failures. Since this is an internal PR, I will fix the failing tests instead of having you address them. I will let you know once I have everything passing and can get this merged

**Comment by cyble-dev** [comment]
Sure @RosenbergYehuda, Thank You.

**Pull request merged**
